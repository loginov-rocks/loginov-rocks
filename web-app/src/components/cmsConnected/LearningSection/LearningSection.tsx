import * as React from 'react';
import { useMemo, useState } from 'react';

import { CmsComponent } from 'cms/interfaces/CmsComponent';
import { CmsConnectedProps } from 'cms/interfaces/CmsConnectedProps';
import { LearningFilter } from 'components/shared/LearningFilter';
import { LearningSectionContext } from 'contexts/LearningSectionContext';

interface Props extends CmsConnectedProps {
  blocks: CmsComponent[];
  title: string;
}

export const LearningSection: React.FC<Props> = ({ blocks, render, title }) => {
  const [selectedYear, setSelectedYear] = useState<string | null>(null);
  const [years, setYears] = useState<string[]>([]);

  const contextValue = useMemo(() => ({
    addCompleted: (completed: string): void => {
      const year = completed.substring(0, 4);

      setYears((prevYears) => {
        if (prevYears.includes(year)) {
          return prevYears;
        }

        return [
          ...prevYears,
          year,
        ];
      });
    },
  }), []);

  const handleSelectYear = (year: string | null): void => {
    setSelectedYear(year);
  };

  return (
    <>

      <h2>{title}</h2>

      <LearningFilter onSelectYear={handleSelectYear} selectedYear={selectedYear} years={years} />

      <LearningSectionContext.Provider value={contextValue}>
        {render(blocks)}
      </LearningSectionContext.Provider>

    </>
  );
};
