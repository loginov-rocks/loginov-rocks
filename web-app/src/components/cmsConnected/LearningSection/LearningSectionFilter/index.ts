import { LearningSectionFilter } from './LearningSectionFilter';

export { LearningSectionFilter };
