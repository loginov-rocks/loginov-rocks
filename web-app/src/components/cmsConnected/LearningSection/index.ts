import { LearningSection } from './LearningSection';

export { LearningSection };
