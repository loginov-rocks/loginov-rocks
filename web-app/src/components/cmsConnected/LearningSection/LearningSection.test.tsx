/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { LearningSection } from './LearningSection';

it('matches snapshot', () => {
  const wrapper = shallow(<LearningSection blocks={[]} render={() => null} title="Learning" />);

  expect(wrapper).toMatchSnapshot();
});
