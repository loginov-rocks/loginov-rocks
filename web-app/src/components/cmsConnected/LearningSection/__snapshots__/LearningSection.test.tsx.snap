// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <h2>
    Learning
  </h2>
  <LearningSectionFilter
    filterYear={null}
    onSelectYear={[Function]}
    years={Array []}
  />
  <ContextProvider
    value={
      Object {
        "addCompleted": [Function],
        "filterYear": null,
      }
    }
  />
</Fragment>
`;
